src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/OffenderControllerTest.kt

```diff
@@ -95,4 +97,23 @@ class OffenderControllerTest {
 
     verify(offenderService, times(1)).updateProbationTeam("exampleCrn", body)
   }
+
+  @Test
+  fun `update offender details`() {
+    val body = UpdateOffenderDetailsRequest(
+      forename = "Alex",
+      middleNames = "Brian Cameron",
+      surname = "David-Edgar",
+      dateOfBirth = LocalDate.parse("1970-01-01"),
+    )
+
+    val request = put("/offender/nomisid/exampleNomisId/update-offender-details")
+      .accept(MediaType.APPLICATION_JSON)
+      .contentType(MediaType.APPLICATION_JSON)
+      .content(mapper.writeValueAsBytes(body))
+
+    mvc.perform(request).andExpect(status().isOk)
+
+    verify(offenderService, times(1)).updateOffenderDetails("exampleNomisId", body)
+  }
```

Comment:
should probably start testing that passing invalid payloads generates appropriate responses